client/lib/user/user.js

```diff
@@ -69,6 +70,9 @@ User.prototype.initialize = function() {
 		if ( this.data ) {
 			this.clearStoreIfChanged( this.data.ID );
 			store.set( 'wpcom_user', this.data );
+			if ( this.data.abtests ) {
```

Comment:
the code in this section looks just about identical to the `handleSuccess` for the user fetch that you've made later in the code.
Are there any differences in the two and if not can we reuse the fn?

Reply:
The only difference seems to be the handling of settings in the `if ( this.settings ) { ... }` block. I'll see if I can refactor this further.